### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by fussel178**
## Added
- Basic functionality
- There are different properties for different unit types.
- You can give additional properties for every unit.
- You can disable the default properties, if you wish. (and add your own)
- basic unit testing
- automatic property conversion

## Notes
- Background process is calling: `systemctl show [unitName] --property [properties]`
- given unit names without a name ending will be interpreted as systemd services

`shell-exec` module unit tests are mocked with:
```js
jest.mock('shell-exec', () => {
    return jest.fn(command => {
        ...
    });
});
```
to enable enhanced test functionality.

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = Object.freeze({
+    PROPERTIES: {
+        any: 'LoadState,ActiveState,SubState,Description,Id,Result,UnitFileState,ActiveEnterTimestamp,ActiveEnterTimestampMonotonic',
+        service: 'CleanResult,MainPID,Type,Restart,MemoryCurrent,Nice,RemainAfterExit,TasksCurrent',
+        mount: 'Where,What,Type,LazyUnmount',
+        slice: 'TasksCurrent'
+        //target: '',
```
Comment (pklaschka) [inline src/constants.js]: Any reason these are commented out, but still there?

File: src/index.js [hunk 2]
```diff
@@ -1,8 +1,74 @@
...
+ * 
+ * @argument options array of systemd units to check
+ * @returns {object|array|string|number|boolean} A JSON-serializable (via `JSON.stringify()`) version information about the server state
  */
-module.exports = async function() {
-    // TODO: Implement module
-    return 'Module loaded';
+module.exports = async function (units) {
+    if (!units || !units.length)
+        throw new Error('No units were specified!');
```
Comment (pklaschka) [inline src/index.js]: Could we change this error message into something a little more "helpful"? E.g., `No units were specified. Expected argument to be of type string[], but got ${typeof units}`...

Also, we should avoid exclamation marks in error messages. They seem unnecessarily "loud"...

**Review by pklaschka — CHANGES_REQUESTED** [review]
Looks good. I've added two comments inline, other than that, everything looks good.

File: src/constants.js (hunk 1 above)
Comment (fussel178) [reply src/constants.js]: Because I want to collect every type of systemd unit for easier editing.

File: src/index.js (hunk 2 above)
Comment (fussel178) [reply src/index.js]: Ok, good point. I will go through every Error message one more time.

File: README.md [hunk 3]
```diff
@@ -6,7 +6,114 @@
 [![Coverage Status](https://coveralls.io/repos/github/server-state/systemd-module/badge.svg?branch=master)](https://coveralls.io/github/server-state/systemd-module?branch=master)
 ![module type: official](https://img.shields.io/badge/module%20type-official-%23015ba0)
 
-A module to view the current status of your systemd services.
+### Description
+
+A module to view the current status of your systemd services. Its response is a object containing keys named as given unit names in [SMF](https://github.com/server-state/specs/blob/master/terminology/server-module-function.md) options with another object as value.
+This object contains the properties of the given unit as key/value pairs, for example:
+```json
```
Comment (pklaschka) [inline README.md:13]: That's `js`. For `json`, you need double quotes around all keys etc. :wink:

File: README.md [hunk 4]
```diff
@@ -6,7 +6,114 @@
...
+    {
+        name: 'user.slice',
+        addProps: 'TasksCurrent',
+        default: false
+    }
+]);
+```
+to your current api server.
+This results in the following output:
+```json
```
Comment (pklaschka) [inline README.md:58]: Same as above...

**Review by pklaschka — DISMISSED** [review]
I'm happy with the JS part now, it's just a problem in your README.md introduced with one of the commits...

File: README.md (hunk 3 above)
Comment (fussel178) [reply README.md:13]: Fixed.

File: README.md (hunk 4 above)
Comment (fussel178) [reply README.md:58]: Fixed.

**Review by pklaschka — APPROVED** [review]

**Pull request merged**